mono/metadata/class-init.c

```diff
@@ -1554,17 +1554,29 @@ count_virtual_methods (MonoClass *klass)
 		mcount = mono_class_get_method_count (klass);
 		for (i = 0; i < mcount; ++i) {
 			flags = klass->methods [i]->flags;
-			if (flags & METHOD_ATTRIBUTE_VIRTUAL)
+			if ((flags & METHOD_ATTRIBUTE_VIRTUAL))
+			{
+				if (!mono_class_is_gtd (klass) && (flags & METHOD_ATTRIBUTE_ABSTRACT  && flags & METHOD_ATTRIBUTE_FINAL)) //reabstraction
```

Comment:
1. Why can't `klass` be a GTD?  Wouldn't this example be ok:
   ```
     interface I1<T> {
        void M1(T t) { Console.WriteLine ("I1.M1");
     }
    interface I2<T> : I1<T> {
       abstract void M1(T t);
    }
    class C<T> : I2<T> { }
    ...
    new C<string> ().M1 ("abcd");
   ``` 

2. I think you should make a helper function that checks for `flags & METHOD_ATTRIBUTE_VIRTUAL && flags & METHOD_ATTRIBUTE_ABSTRACT  && flags & METHOD_ATTRIBUTE_FINAL` since it's used in a couple of places and it will be easier to think about it if it was called `mono_method_is_reabstracted`